build: git remote authentication secrets (#23460)
docker/docker.github.io: content/manuals/build/building/secrets.md

For example, say you have a private GitHub repository at
`https://github.com/example/todo-app.git`, and you want to run a build using
that repository as the build context. An unauthenticated `docker build` command
fails because the builder isn't authorized to pull the repository:

```console
$ docker build https://github.com/example/todo-app.git
[+] Building 0.4s (1/1) FINISHED
 => ERROR [internal] load git source https://github.com/example/todo-app.git
------
 > [internal] load git source https://github.com/example/todo-app.git:
0.313 fatal: could not read Username for 'https://github.com': terminal prompts disabled
------
```

To authenticate the builder to GitHub, set the `GIT_AUTH_TOKEN`
environment variable to contain a valid GitHub access token, and pass it as a
secret to the build:

```console
$ GIT_AUTH_TOKEN=$(gh auth token) docker build \
  --secret id=GIT_AUTH_TOKEN \
  https://github.com/example/todo-app.git
```

The `GIT_AUTH_TOKEN` also works with `ADD` to fetch private Git repositories as
part of your build:

```dockerfile
FROM alpine
ADD https://github.com/example/todo-app.git /src
```

### HTTP authentication scheme

> BuildKit supports two types of Git authentication secrets, and you should use either one or the other, not both:

Reviewer: ```suggestion
BuildKit supports two Git authentication secrets:
```

> - **`GIT_AUTH_TOKEN`**: Uses Basic authentication with a fixed username of `x-access-token` (GitHub-specific)

Reviewer: ```suggestion
- **`GIT_AUTH_TOKEN`**: Uses Basic authentication with a fixed username of `x-access-token` (the GitHub-style default)
```

- **`GIT_AUTH_HEADER`**: Uses the raw authorization header value you provide (works with any Git provider)

> #### Using GIT_AUTH_TOKEN (f.ex. GitHub)

Reviewer: ```suggestion
#### Using GIT_AUTH_TOKEN (for example, GitHub)
```

> When you use `GIT_AUTH_TOKEN`, BuildKit automatically constructs a Basic authentication header using `x-access-token` as the user:

Reviewer: ```suggestion
When you use `GIT_AUTH_TOKEN`, BuildKit constructs a Basic authentication header using `x-access-token` as the user:
```

```http
Authorization: Basic <base64("x-access-token:<GIT_AUTH_TOKEN>")>
```

> This method works with GitHub. Example usage:

Reviewer: ```suggestion
This method works for providers that accept the `x-access-token` Basic auth pattern, such as GitHub. Example usage:
```

```console
$ export GIT_AUTH_TOKEN=$(gh auth token)
$ docker build \
  --secret id=GIT_AUTH_TOKEN \
  https://github.com/example/todo-app.git
```

> #### Using GIT_AUTH_HEADER (Any Git provider)

Reviewer: ```suggestion
#### Using GIT_AUTH_HEADER (custom authorization header)
```

> When you use `GIT_AUTH_HEADER`, BuildKit uses the exact value you provide as the authorization header:

Reviewer: ```suggestion
When you use `GIT_AUTH_HEADER`, BuildKit uses the exact value you provide as the `Authorization` header:
```
